analyzers/tests/SonarAnalyzer.UnitTest/Extensions/PropertyDeclarationSyntaxExtensionsTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Linq;
+using FluentAssertions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarAnalyzer.Extensions;
+
+namespace SonarAnalyzer.UnitTest.Extensions
+{
+    [TestClass]
+    public class PropertyDeclarationSyntaxExtensionsTests
+    {
+        [TestMethod]
+        public void IsAutoProperty_AccessorWithBody_ReturnsFalse()
+        {
+            const string code = @"class TestCases
+{
+    public int Property { get { return 0; } }
+}
+";
+            GetPropertyDeclaration(code).IsAutoProperty().Should().BeFalse();
+        }
+
+        [TestMethod]
+        public void IsAutoProperty_AccessorWithExpressionBody_ReturnsFalse()
+        {
+            const string code = @"record TestCases
+{
+    public int Property
+    {
+        get => 0;
```

Comment:
could you also add a test with an implemented `set` and `init` ?

Reply:
As far as I can tell this test would be redundant since all accessors are treated the same; no matter their type, we check if they have a body or expression body and this case is covered by `IsAutoProperty_AccessorWithBody_ReturnsFalse`

I prefer not to add this test.

Reply:
You are testing thinking of the implementation instead of the functionality...
If someone will come and change to only test getters, should a test fail?

Reply:
As far as I can see, it is a compromise. In general checking all possible cases take more time to run (running more cases) and more time to maintain the tests so this comes with a cost we should not ignore.

From the other point of view. if the code is changed but not covered by tests we will have low coverage and we will have to do something anyway. Our threshold is 95% which is sensible enough to spot this kind of changes in most of the cases.

Reply:
In the case that I mentioned, there will be no change in code coverage: only getters are tested, so if someone changes the code to test only the getters, coverage will be 100%.

Reply:
As discussed also offline, if we change the code to consider only getters, most probably we will do an equality check which will create a logic branch uncovered by tests.